pynicotine/utils.py

```diff
@@ -204,6 +204,37 @@ def unescape(string):
     return string
 
 
+def find_whole_word(word, text):
+    """Returns start position of a whole word that is not in a subword."""
+
+    if word not in text:
+        return -1
+
+    word_boundaries = [" "] + PUNCTUATION
+    whole = False
+    start = after = 0
+
+    while not whole and start > -1:
+        start = text.find(word, after)
+        after = start + len(word)
+
+        whole = ((text[after] if after < len(text) else " ") in word_boundaries
+                 and (text[start - 1] if start > 0 else " ") in word_boundaries)
+
+    return start if whole else -1
+
+
+def censor_text(text, censored_patterns):
```

Comment:
`def censor_text(text, censored_patterns, filler="*"):`

It might be desirable to set a custom filler character, such as from a plugin or such.